packages/api/tests/animal_type.test.js

```diff
@@ -0,0 +1,127 @@
+/*
+ *  Copyright (c) 2024 LiteFarm.org
+ *  This file is part of LiteFarm.
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
+import chai from 'chai';
+import util from 'util';
+
+import chaiHttp from 'chai-http';
+chai.use(chaiHttp);
+
+import server from './../src/server.js';
+import knex from '../src/util/knex.js';
+import { tableCleanup } from './testEnvironment.js';
+
+jest.mock('jsdom');
+jest.mock('../src/middleware/acl/checkJwt.js', () =>
+  jest.fn((req, res, next) => {
+    req.auth = {};
+    req.auth.user_id = req.get('user_id');
+    next();
+  }),
+);
+import mocks from './mock.factories.js';
+
+describe('Animal Type Tests', () => {
+  let token;
+  let farm;
+  let newOwner;
+
+  beforeAll(() => {
+    token = global.token;
+  });
+
+  function getRequest({ user_id = newOwner.user_id, farm_id = farm.farm_id }, callback) {
+    chai
+      .request(server)
+      .get('/animal_type')
+      .set('user_id', user_id)
+      .set('farm_id', farm_id)
+      .end(callback);
+  }
+
+  const getRequestAsPromise = util.promisify(getRequest);
+
+  function fakeUserFarm(role = 1) {
+    return { ...mocks.fakeUserFarm(), role_id: role };
+  }
+
+  async function returnUserFarms(role) {
+    const [mainFarm] = await mocks.farmFactory();
+    const [user] = await mocks.usersFactory();
+
+    await mocks.userFarmFactory(
+      {
+        promisedUser: [user],
+        promisedFarm: [mainFarm],
+      },
+      fakeUserFarm(role),
+    );
+    return { mainFarm, user };
+  }
+
+  async function makeUserCreatedAnimalType(mainFarm) {
+    const [animal_type] = await mocks.animal_typeFactory({
+      promisedFarm: [mainFarm],
+    });
+    return animal_type;
+  }
+
+  beforeEach(async () => {
+    [farm] = await mocks.farmFactory();
+    [newOwner] = await mocks.usersFactory();
+  });
+
+  afterAll(async (done) => {
+    await tableCleanup(knex);
+    await knex.destroy();
+    done();
+  });
+
+  // GET TESTS
+  describe('GET animal type tests', () => {
+    test('All users should get animal type by farm id (or null)', async () => {
```

Comment:
Working on tests super similar to these ones on my branch, I realized in this test we would be creating a custom animal type, but there aren't any default types necessarily, so we wouldn't be truly asserting that the types with farm ID null are actually being returned. In my PR I added something to populate a default type, maybe we could leverage that here too and ensure the response is the right length?

Reply:
Thank you for writing those! I did run it locally with the defaults still present but of course yes, in CI there is no guarantee they are still there and I let it be only out of laziness 🙂 I do wish we had a better `tableCleanup` that didn't destroy the migration-created defaults.